Bug fix tank volume

ssankalp26 opened the pull request:
1. Fixed bugs in existing code for tank volume computation
2. Added support for aft tank volume computations
3. Now requires additional package, "shapely" to support aft tank volume computations

ssankalp26 committed:
adding try and except for tank if insufficicnet space in existing segment

ssankalp26 committed:
variable. to store total fuel volume

ssankalp26 committed:
added a pop fuel tanks, so it doesnt get repeated

ssankalp26 committed:
Revert "added a pop fuel tanks, so it doesnt get repeated"

This reverts commit 5c34dbad53f3f0a47f01beefb0ff377743d4ef58.

mclarke2 committed:
updated code for Rohit

ssankalp26 committed:
Revert "updated code for Rohit"

This reverts commit 898288e404f10089d19bff6e5a15ef1c7e0d279d.

ssankalp26 committed:
working aft transformed tank sizing script and optimization

Will refactor code to make it more efficient

mclarke2 committed:
Update Non_Integral_Tank.py

mclarke2 committed:
Update plot_3d_fuel_tank.py

mclarke2 committed:
updated code

ssankalp26 committed:
Add offset to non-integral fuel tank calculations and corrected hemisphere

Introduces an 'offset' attribute to Non_Integral_Tank and updates fuel volume and 3D plotting logic to account for this offset.

ssankalp26 committed:
fix to aft tank modelling

The sides tank origins are off

ssankalp26 committed:
updates

ssankalp26 committed:
Integral Tank Clean up

ssankalp26 committed:
Refactoring tank volume computation

ssankalp26 committed:
Update pyproject.toml

ssankalp26 committed:
Update fuel_tank_volume_test.py

ssankalp26 committed:
reduce error tol

ssankalp26 committed:
Updates airfoils to be txt files

ssankalp26 committed:
Update test_automatic_regression.py

Unncomment the rest of the tests for CI

ssankalp26 committed:
fixed for predefined tanks

ssankalp26 committed:
Update blended_wing_body_aerodynamics_test.py

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/__init__.py:
```diff
@@ -0,0 +1,18 @@
+# RCAIDE/Methods/Powertrain/Sources/Fuel_Tanks/__init__.py
+# 
+
+""" REDOOOO
```
Remove REDOOO

ssankalp26 replied:
Done

mclarke2 commented on RCAIDE/Library/Components/Powertrain/Sources/Fuel_Tanks/Fuel_Tank.py:
```diff
@@ -95,4 +96,8 @@ def compute_tank_properties(self,state,fuel_line):
+        return                           
+    
+    def compute_volume(self,wings,fuselages):
+        volume =  self.internal_volume
```
thos function seems pointless

ssankalp26 replied:
You are right This function in itself does not do anything. However this function is required for when the the "compute_volume" is called. Also Since this is a parent class it is kinda rudimentary

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -0,0 +1,143 @@
+    return volume
+
+def compute_wing_integral_tank_volume(fuel_tank,wing):
+
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -0,0 +1,143 @@
+
+
+def compute_fuselage_integral_tank_fuel_volume(fuel_tank,fuselage):
+    if len(fuselage.segments) > 1:
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -0,0 +1,143 @@
+
+
+def compute_wing_integral_tank_fuel_volume(fuel_tank,wing):     
+
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -0,0 +1,143 @@
+    return volume
+
+def compute_non_dimensional_rib_coordinates(compoment): 
+    if compoment.airfoil != None: 
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -0,0 +1,143 @@
+
+
+def compute_segmented_wing_integral_tank_fuel_volume(fuel_tank,wing,inner_segment,outer_segment):   
+
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/__init__.py:
```diff
@@ -0,0 +1,18 @@
+# RCAIDE/Methods/Powertrain/Sources/Fuel_Tanks/__init__.py
+# 
+
+""" REDOOOO
```
Remove

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -0,0 +1,390 @@
+# ----------------------------------------------------------------------------------------------------------------------  
+
+def compute_bwb_aft_tank_volume(fuel_tank,wing):
+
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -0,0 +1,390 @@
+    return volume
+
+def compute_wing_non_integral_tank_volume(fuel_tank,wing):
+    if len(wing.segments) > 1: 
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -0,0 +1,390 @@
+    return volume
+
+def compute_wing_non_integral_tank_fuel_volume(fuel_tank,wing,inner_segment_0,outer_segment,tank_percent_span_location):
+
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -0,0 +1,390 @@
+
+
+def compute_largest_circle(x_points,z_upper,z_lower):
+
```
add a header to this function

ARmolloy replied:
Done

ssankalp26 replied:
Done by @Aidan

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -0,0 +1,390 @@
+                best_r = r
+                best_pt = (x, z)
+    max_diameter = 2 * best_r
+    ########Delete Later########################################################################################################
```
Delete

ssankalp26 replied:
Done

mclarke2 commented on VnV/Vehicles/Airfoils/main_wing_airfoil_XSec_0.txt:
```diff
@@ -0,0 +1,35 @@
+main_wing_airfoil_XSec_0.txt
```
Remove this proprietary airfoils

ssankalp26 replied:
Done

mclarke2 commented on VnV/Verification/geometry/fuel_tank_volume_test.py:
```diff
@@ -1,7 +1,7 @@
-# Regression/scripts/Tests/emissions_test.py
-# (c) Copyright 2023 Aerospace Research Community LLC
+# Regression/scripts/Tests/o//////////////.py
```
?

ssankalp26 commented on RCAIDE/Library/Methods/Geometry/Planform/compute_fuel_volume.py:
```diff
@@ -23,295 +27,12 @@ def compute_fuel_volume(vehicle, update_max_fuel =True):
- 
-
-
+                volume = fuel_tank.compute_volume(wings,fuselages)
```
This is why I have the compute volume in the FUel Tank Class.

ssankalp26 commented on VnV/Verification/geometry/fuel_tank_volume_test.py:
```diff
@@ -1,7 +1,7 @@
+#
 # 
-# Created:  Jul 2023, M. Clarke 
+# Created:  J
```
Done

ssankalp26 committed:
updates to improv coverage and comments addresed

function headers to be addressed by @ARmolloy

ARmolloy committed:
Documentation

ssankalp26 committed:
airfoil updates for bwb

ssankalp26 committed:
Relax assertion tolerance in fuel tank volume test

The acceptable error threshold in non_integral_fuel_tank_volume_test was increased from 1e-6 to 1e-2 to allow for greater numerical tolerance in the test results.
